ci/jenkins/Test.groovy.j2

```diff
@@ -217,7 +234,7 @@ stage('Test') {
     )
   {% endcall %}
   {% call m.test_step(
-    name="test: QEMU",
+    name="test: QEMU CPP & Demo",
```

Comment:
Since the demos and C++ tests only take about 3 minutes each I think we can keep them on the first qemu shard (i.e. similar to how there are `{% if shard_index == 1 %}` on some other tests)